docs/design.md

```diff
@@ -0,0 +1,67 @@
+# Design
+
+## Models
+
+A Model object represents a logical ML model.
+
+The object describes where the source code for the model should come from. This code will be stored in a container image. That container image will initially be run in order to store the initial weights and biases in a bucket that is mounted into the container as a directory. This is most useful in cases where the Model is representing a pre-trained model.
+
+The Model object can optionally describe the training parameters including a reference to the Dataset that should be used for training. In this case, the Model object represents the eventual output of a training/finetuning operation.
+
+```yaml
+kind: Model
+spec:
+  source: {} # Required
+  training: {} # Optional, will run a training job
+```
+
+### Model Sources
+
+A Model can be sourced from Git. This will trigger a build Job.
+
+```yaml
+kind: Model
+spec:
+  source:
+    git:
+      url: https://github.com/substratusai/model-falcon-7b
```

Comment:
I guess in this case it could be 
```
url: https://github.com/substratusai/model-loader-huggingface
name: falcon-7b
```

The huggingface model loader can load any model from huggingface into the GCS bucket